Source/CombatExtended/CombatExtended/Verbs/Verb_ShootMortarCE.cs

```diff
@@ -53,5 +253,60 @@ protected virtual float GetMissRadiusForDist(float targDist)
             float missRadiusFactor = rangePercent <= 0.5f ? 1 - rangePercent : 0.5f + ((rangePercent - 0.5f) / 2);
             return VerbPropsCE.circularError * missRadiusFactor;
         }
+
+        protected virtual float GetGlobalMissRadiusForDist(float targDist)
+        {
+            float maxRange = this.projectilePropsCE.shellingProps.range * 5f;
+            float rangePercent = targDist / maxRange;
+            float missRadiusFactor = rangePercent <= 0.5f ? 1 - rangePercent : 0.5f + ((rangePercent - 0.5f) / 2);
+            return VerbPropsCE.circularError * missRadiusFactor;
+        }
+
+        private void ShiftGlobalTarget(ShiftVecReport report)
+        {
+            if (report == null || !shiftedGlobalCell.IsValid)
+            {
+                return;
+            }
+            float minDistFactor = !targetHasMarker ? 0.444f : 0.111f;
+            report.shotDist = Mathf.Max(report.shotDist, report.maxRange * minDistFactor);
+            float shotDist = report.shotDist;
+            var target = new Vector2(shiftedGlobalCell.x, shiftedGlobalCell.z);
+            var direction = (Find.WorldGrid.GetTileCenter(startingTile) - Find.WorldGrid.GetTileCenter(destinationTile)).normalized;
+
+            var estimatedTargDist = report.GetRandDist();
+            var spreadVec = UnityEngine.Random.insideUnitCircle * Mathf.Clamp(report.spreadDegrees * Mathf.PI / 360f, -1f, 1f) * (estimatedTargDist - report.shotDist);
+            var ray = new Ray2D(target, -1 * direction);
+            var missVec = report.GetRandCircularVec();
+            var shiftedTarg = ray.GetPoint((estimatedTargDist - shotDist)) + missVec + spreadVec;
+
+            shiftedGlobalCell = new IntVec3((int)shiftedTarg.x, 0, (int)shiftedTarg.y);
+        }
+
+        //private void CanHitFlyOverRoofTarget(IntVec3 source, IntVec3 destination)
```

Comment:
A big commented-out section. Looks like it was initially intended to prevent artillery from being dropped on mountain tiles? Do we need to keep it for later, or can it be cleaned up?

Reply:
That code made by kbatbouta.
Not sure that's really needed.
Also, I think, using that code potentially can break sieges:
There will be no targets for sieges when player using cave-bases.

I guess it can be cleaned up.